README.md

```diff
@@ -13,66 +13,113 @@ Assumptions:
+    - _NOTE: Default env var values are set in the `Dockerfile` and will be used in `docker run` if none are provided_
+2. Using `export` to set them in the current executing shell environment.
+
+    ```bash
+    $ export USAPENDING_API=http://localhost:8000/api
+    $ npm dev
+    ```
+    - `docker-compose` will also pick these up, but `docker run` will not and fall back to `Dockerfile` defaults.
+3. On the command-line, prior to a shell command, to inject into the sub-shell environment
+
+    ```bash
+    $ USAPENDING_API=http://localhost:8000/api npm dev
+    ```
+
+The latter methods in this list take precedence over environment variables set in an earlier method. 
+- If you build or run `npm` scripts on your local desktop development environment with the `npm` CLI, or switch between this and Docker, setting up your shell with `export` (or a tool like `direnv`) is most flexible. 
+- If you only run scripts/builds in Docker or Docker Compose, using a `.env` file is a simple, clean, config-file-driven approach 
+
+### Scripts
+_Custom and life-cycle scripts to execute, as defined under the `scripts` property in `package.json`_
+
+| Script       | Output                                                      |
+|--------------|-------------------------------------------------------------|
+| `npm start`  | dev server with hot reloading                               |
+| `npm prod`   | generates static assets w/ production optimization          |
+| `npm dev`    | generates static assets w/ development optimization         |
+| `npm lint`   | executes linter                                             |
+| `npm test`   | executes unit tests                                         |
+| `npm sass`   | dev server w/ scss source maps                              |
+| `npm travis` | executes travis validation script                           |
+| `npm ci`     | clean existing Node dependencies and install dependencies   |
+
+### Bundling and Deployment
+_Bundling and running the website_
+
+To keep a clean workspace, our CI/CD pipelines use Docker to gather Node dependencies, bundle artifacts, and deploy them. Developers can similarly do this in their local development environments to reduce tool install/versioning and dependency management.
+
+To run the website _**locally**_ with pre-bundled artifacts, first download NPM packages, then generate and bundle static artifacts, then run an `nginx` container to host them. You can do this all in _one step_ with just:
+```bash
+docker-compose build && docker-compose run --rm bundle-usaspending-website && docker-compose up usaspending-website
+``` 
+- _The first two steps can take 5+ minutes each, so this could be a total of ~10 minutes before ready._
+
+To do this in individual, sequential steps, from the `usaspending-website` source root:
+1. Ensure your `usaspending-website` Docker image has the latest dependencies:
+
+    ```bash
+    docker-compose build
```

Comment:
can we just use docker build here instead of compose? It might be useful to distinguish between:

(a) docker-compose workflow (one step, as defined above) and 
(b) docker run work flow (multiple steps)

Please correct me if I'm wrong wrong about this but it seems each step below could just as well use docker run... rather than docker-compose...?

Reply:
But that's the gist of this PR ... to get rid of those low-level details:
> Favoring hiding away docker specifics (ports, volumes, env vars) into a docker-compose.yml file for easier use

Using `docker-compose` makes the commands much simpler and hides away
- the volume for getting source files
- the ports inside and outside the container
- the name of the image 
- the volume for mounting artifacts to nginx
- the version of nginx to use
- the fact we use an nginx container
- etc.